src/renderer/routes/test_sequencer_panel/components/DesignBar.tsx

```diff
@@ -0,0 +1,266 @@
+      elem.status != "running"
+    )
+      count++;
+  });
+  return count;
+};
+
+const getNumberOfSequence = (data: TestSequenceContainer[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (elem.runable) count++;
+  });
+  return count;
+};
+
+const getNumberOfSequenceRun = (data: TestSequenceContainer[]): number => {
+  let count = 0;
+  data.forEach((elem) => {
+    if (elem.runable && elem.status != "pending" && elem.status != "running")
+      count++;
+  });
+  return count;
+};
+
+export const getGlobalStatus = (
+  cycleRuns: TestSequenceContainer[][],
+  sequences: TestSequenceContainer[],
+  data: TestSequenceElement[],
+): StatusType => {
+  // Create a priority list with all the status
+  const priority = {
+    pending: 0,
+    pass: 1,
+    fail: 2,
+    aborted: 3,
+    paused: 4,
+    running: 5,
+  };
+  // Find highest priority in cycle
+  const highestCycle =
+    cycleRuns.length > 0
+      ? cycleRuns
+          .map((el) => getGlobalStatus([], el, []))
+          .reduce((prev, curr) =>
+            priority[prev] > priority[curr] ? prev : curr,
+          )
+      : "pending";
+  if (sequences.length === 0 && data.length === 0) return highestCycle;
+  // Highest in the view
+  const iter = sequences.length > 0 ? sequences : data;
+  const status = iter
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    .map((el: any) => el.status)
+    .reduce((prev, curr) => (priority[prev] > priority[curr] ? prev : curr));
+  const highestGlobal =
+    priority[highestCycle] > priority[status] ? highestCycle : status;
+  return highestGlobal;
+};
+
+const mapStatusToDisplay: { [k in StatusType] } = {
+  pending: (
+    <div className="inline-flex h-full w-full items-center justify-center rounded-md bg-secondary text-primary">
+      <code data-testid="global-status-badge">PENDING</code>
+    </div>
+  ),
+  running: (
+    <div className="inline-flex h-full w-full items-center justify-center rounded-md bg-blue-500 text-primary-foreground">
+      <code data-testid="global-status-badge">RUNNING</code>
+    </div>
+  ),
+  paused: (
+    <div className="inline-flex h-full w-full items-center justify-center rounded-md bg-yellow-500 text-primary-foreground">
+      <code data-testid="global-status-badge">PAUSED</code>
+    </div>
+  ),
+  pass: (
+    <div className="inline-flex h-full w-full items-center justify-center rounded-md bg-green-500 text-primary-foreground">
+      <code data-testid="global-status-badge">PASS</code>
+    </div>
+  ),
+  aborted: (
+    <div className="inline-flex h-full w-full items-center justify-center rounded-md bg-red-500 text-primary-foreground">
+      <code data-testid="global-status-badge">ABORTED</code>
+    </div>
+  ),
+  fail: (
+    <div className="inline-flex h-full w-full items-center justify-center rounded-md bg-red-500 text-primary-foreground">
+      <code data-testid="global-status-badge">FAIL</code>
+    </div>
+  ),
+};
```

Comment:
this could be a lot shorter by turning it into its own component and cutting out a lot of the repeated css (lookup `class-variance-authority`, it's already installed in studio)